use a global list for toasts

song50119 opened the pull request:
Resolves #425

Commit:
use a global list for toasts

oshorefueled commented on ui/page.go:
```diff
@@ -112,8 +112,8 @@ type pageCommon struct {
 	keyEvents       chan *key.Event
-	toast           chan *toast
-	toastLoad       *toast
+	toasts          *[]*toast
```
Rather than using a pointer to a slice, make the method on the pageCommon struct to use a pointer receiver. Change the other methods to pointer receivers as well to maintain consistency.

oshorefueled commented on ui/page.go:
```diff
@@ -299,8 +297,11 @@ func (win *Window) addPages(decredIcons map[string]image.Image) {
 		changePage:              win.changePage,
 		setReturnPage:           win.setReturnPage,
 		refreshWindow:           win.refresh,
+		toastList:               layout.List{Axis: layout.Vertical},
```
I don't think godcr has a use case where there will be more than one notification at once. That means there's no need to create a list layout.

song50119 replied:
If only display toast at once app only need single struct of toast, I don't think app need slice of toasts

oshorefueled commented on ui/toast.go:
```diff
@@ -37,14 +37,19 @@ func displayToast(th *decredmaterial.Theme, gtx layout.Context, n *toast) layout
+// Timer is only creates a new time.Timer when toast.timer is nil.
 // Gio re-renders its UI recursively and Timer prevents multiple time.Timer from being created before the duration is
 // exceeded.
 func (n *toast) Timer(d time.Duration, f func()) {
```
This method can be removed. Using time.AfterFunc would be sufficient. The entire toast.go file can be deleted and `displayToast` method can be added to components.go

oshorefueled commented on ui/page.go:
```diff
@@ -473,37 +472,27 @@ func (page pageCommon) Layout(gtx layout.Context, body layout.Widget) layout.Dim
-				page.toastLoad.Timer(time.Second*3, func() {
-					page.toastLoad.text = ""
+					}
+					return t((*page.toasts)[index])
```
You can simply loop through the list of toasts, call the time.AfterFunc method to remove that toast from the list in the specified duration. You should add a mutex lock for appending and removing data from the toast list so a data race issue doesn't occur when trying to access data in the list.

Commit:
Update toast
- Display only once toast
- Remove toast.go file and move the toast widget to component

oshorefueled commented on ui/compontents.go:
```diff
@@ -940,7 +948,45 @@ func ticketCard(gtx layout.Context, c pageCommon, t *wallet.Ticket) layout.Dimen
+}
+
+func (page pageCommon) handleToast() {
+	if (*page.toast) == nil {
+		return
+	}
+
+	if (*page.toast).timer == nil {
+		(*page.toast).timer = time.NewTimer(time.Second * 3)
+	}
+
+	select {
+	case <-(*page.toast).timer.C:
+		*page.toast = nil
+	default:
```
This can be replaced with time.AfterFunc
